azurerm/resource_arm_container_group.go

```diff
@@ -564,10 +571,24 @@ func expandContainerGroupContainers(d *schema.ResourceData) (*[]containerinstanc
 			containerGroupPorts = append(containerGroupPorts, containerGroupPort)
 		}
 
+		// Set both secure and non-secure environment variables
+		var evar *[]containerinstance.EnvironmentVariable
+		var secevar *[]containerinstance.EnvironmentVariable
+
 		if v, ok := data["environment_variables"]; ok {
-			container.EnvironmentVariables = expandContainerEnvironmentVariables(v)
+			evar = expandContainerEnvironmentVariables(v, false)
+		}
+
+		if v, ok := data["secure_environment_variables"]; ok {
+			secevar = expandContainerEnvironmentVariables(v, true)
 		}
 
+		for _, v := range *secevar {
+			*evar = append(*evar, v)
+		}
```

Comment:
Allocation for too many times, change it to:

evar = &append(*evar, *secevar...)